src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -376,6 +376,226 @@ bool LibraryCallKit::inline_vector_nary_operation(int n) {
+  }
+
+  Node* opd1 = NULL; Node* opd2 = NULL; Node* opd3 = NULL;
+  switch (n) {
+    case 3: {
+      opd3 = unbox_vector(argument(7), vbox_type, elem_bt, num_elem);
+      if (opd3 == NULL) {
+        if (C->print_intrinsics()) {
+          tty->print_cr("  ** unbox failed v3=%s",
+                        NodeClassNames[argument(7)->Opcode()]);
+        }
+        return false;
+      }
+      // fall-through
+    }
+    case 2: {
+      opd2 = unbox_vector(argument(6), vbox_type, elem_bt, num_elem);
+      if (opd2 == NULL) {
+        if (C->print_intrinsics()) {
+          tty->print_cr("  ** unbox failed v2=%s",
+                        NodeClassNames[argument(6)->Opcode()]);
+        }
+        return false;
+      }
+      // fall-through
+    }
+    case 1: {
+      opd1 = unbox_vector(argument(5), vbox_type, elem_bt, num_elem);
+      if (opd1 == NULL) {
+        if (C->print_intrinsics()) {
+          tty->print_cr("  ** unbox failed v1=%s",
+                        NodeClassNames[argument(5)->Opcode()]);
+        }
+        return false;
+      }
+      break;
+    }
+    default: fatal("unsupported arity: %d", n);
+  }
+
+  Node* mask = NULL;
+  bool use_predicate = false;
+  if (is_masked_op) {
+    ciKlass* mbox_klass = mask_klass->const_oop()->as_instance()->java_lang_Class_klass();
+    assert(is_vector_mask(mbox_klass), "argument(2) should be a mask class");
+    const TypeInstPtr* mbox_type = TypeInstPtr::make_exact(TypePtr::NotNull, mbox_klass);
+    mask = unbox_vector(argument(n + 5), mbox_type, elem_bt, num_elem);
+    if (mask == NULL) {
+      if (C->print_intrinsics()) {
+        tty->print_cr("  ** unbox failed mask=%s",
+                      NodeClassNames[argument(n + 5)->Opcode()]);
+      }
+      return false;
+    }
+
+    // Return true if current platform has implemented the masked operation with predicate feature.
+    use_predicate = sopc != 0 && Matcher::match_rule_supported_vector_masked(sopc, num_elem, elem_bt);
+    if (!use_predicate && !arch_supports_vector(Op_VectorBlend, num_elem, elem_bt, VecMaskUseLoad)) {
+      return false;
+    }
+  }
+
+  Node* operation = NULL;
+  if (opc == Op_CallLeafVector) {
+    assert(UseVectorStubs, "sanity");
+    operation = gen_call_to_svml(opr->get_con(), elem_bt, num_elem, opd1, opd2);
+    if (operation == NULL) {
+      if (C->print_intrinsics()) {
+        tty->print_cr("  ** svml call failed");
+      }
+      return false;
+     }
+  } else {
+    const TypeVect* vt = TypeVect::make(elem_bt, num_elem);
+    switch (n) {
+      case 1:
+      case 2: {
+        operation = VectorNode::make(sopc, opd1, opd2, vt);
+        break;
+      }
+      case 3: {
+        operation = VectorNode::make(sopc, opd1, opd2, opd3, vt);
+        break;
+      }
+      default: fatal("unsupported arity: %d", n);
+    }
+  }
+
+  if (is_masked_op && mask != NULL) {
+    if (use_predicate) {
+      // TODO: add predicate implementation for masked operation.
```

Comment:
Does it make sense to fail the intrinsification or fail-fast with a crash here until the support is there?